src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -364,6 +388,53 @@ def run(self) -> None:
             self.log.fatal("Fatal runtime error: {}\n{}".format(
                 ex, traceback.format_exc()))
 
+    def get_health_checks(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary of health checks based on the synchronization status of images.
+        """
+        unsynced_images = []
+        total_count = 0
+
+        for next_timestamp in self.queue:
+            for image_spec in self.queue[next_timestamp]:
+                next_snapshot = datetime.strptime(next_timestamp, '%Y-%m-%d %H:%M:%S')
+                with self.module.rados.open_ioctx2(int(image_spec.pool_id)) as ioctx:
+                    image_schedule = self.schedules.find(*image_spec)
+                    if not image_schedule:
+                        continue
+                    min_interval = min(image_schedule.items,
+                                       key=lambda x: x[0].minutes)[0].minutes
+                    try:
+                        with rbd.Image(ioctx, image_id=image_spec.image_id,
+                                       read_only=True) as image:
+                            total_count += 1
```

Comment:
This can cause performance issues as opening an Image i pretty expensive ~5ms per image, so roughly 1000 images will take 5 seconds to parse.

Reply:
NOTE: https://github.com/ceph/ceph/pull/49761 this pr reduces the overhead to ~1.5ms per image.